### ISSUE TITLE
BetterC runtime

### ISSUE THREAD

**Issue by burner** [dlang/project-ideas#18]
[BetterC](https://dlang.org/blog/2017/08/23/d-as-a-better-c) is a subset
of D which limits the feature set of D, s.t. it\'s even easier to
migrate from C as no Garbage Collection (GC) is used anymore. While,
this feature is popular for some projects, \`-betterC\` today means
still means a severe restriction in the amount of D features available
(see the [BetterC specification](https://dlang.org/spec/betterc.html)).
This project is about making more parts of the D runtime
betterC-compatible. For example, classes could be allocated with
reference-counting instead of using the GC.

**Comment by JinShil** [comment]
Related:  https://github.com/JinShil/utiliD, https://github.com/caraus-ecms/tanya

**Comment by JinShil** [comment]
I think instead of a BetterC runtime we should be striving for the opt-in continuum described here:  https://forum.dlang.org/post/q7j4sl$17pe$1@digitalmars.com

**Comment by RazvanN7** [comment]
Yes, this project is analogous with the druntime hooks one: https://github.com/dlang/project-ideas/issues/25

**Comment by PetarKirov** [comment]
It depends - would a fully templated druntime work in kernel-space or embedded platforms? #25 is a necessary tool to accomplish our goals,  but I don't is sufficient by itself. There's additional work that needs to be done to ensure that the templated hooks are usable in a betterC and freestanding (baremetal) environments.
